libtelemetry/src/main/java/com/mapbox/android/telemetry/CertificateBlacklist.java

```diff
@@ -0,0 +1,220 @@
+package com.mapbox.android.telemetry;
+
+import android.content.Context;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageManager;
+import android.util.Log;
+
+import com.google.gson.Gson;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.HttpUrl;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+class CertificateBlacklist implements Callback {
+  private static final String LOG_TAG = "MapboxBlacklist";
+  private static final String BLACKLIST_FILE = "MapboxBlacklist";
+  private static final String SHA256 = "sha256/";
+  private static final String NEW_LINE = "\n";
+  private static final String HTTPS = "https://";
+  private static final String BACKSLASH = "/";
+  private static final String EMPTY_STRING = "";
+  private static final int BLACKLIST_HEAD = 0;
+  private static final String COM_CONFIG_ENDPOINT = "https://api.mapbox.com/events-config";
+  private static final String CHINA_CONFIG_ENDPOINT = "https://api.mapbox.cn/events-config";
+  private static final String ACCESS_TOKEN_QUERY_PARAMETER = "access_token";
+  private static final String ENDPOINT_DETERMINATION_FAIL = "Endpoint Determination Failed";
+  private static final String SAVE_BLACKLIST_FAIL = "Unable to save blacklist to file";
+  private static final String RETRIEVE_TIME_FAIL = "Unable to retrieve last update time from blacklist";
+  private static final String RETRIEVE_BLACKLIST_FAIL = "Unable to retrieve blacklist contents from file";
+  private static final String REQUEST_FAIL = "Request failed to download blacklist";
+  private static final String HTTPS_SCHEME = "https";
+  private static final Map<Environment, String> ENDPOINTS = new HashMap<Environment, String>() {
+    {
+      put(Environment.COM, COM_CONFIG_ENDPOINT);
+      put(Environment.CHINA, CHINA_CONFIG_ENDPOINT);
+    }
+  };
+  private Context context;
+  private String accessToken;
+
+  CertificateBlacklist(Context context, String accessToken) {
+    this.context = context;
+    this.accessToken = accessToken;
+  }
+
+  List<String> retrieveBlackList() {
```

Comment:
What are some ways to make this method content more readable? One potential improvement i see here: have less nested levels of conditions.